### DISCUSSION TITLE
FitnessTracker.authorize(permissions) returns always false for Android

### DISCUSSION THREAD

**Discussion by pandey21197** [doville/rn-fitness-tracker#111] [Q&A] [answered]
FitnessTracker.authorize always return false even though the permission was granted. 
```
    const permissions = {
      healthReadPermissions: [
        HealthKitDataType.StepCount,
        // HealthKitDataType.DistanceWalkingRunning,
      ],
      googleFitReadPermissions: [
        GoogleFitDataType.Steps,
        // GoogleFitDataType.Distance,
      ],
    };

    const onReadSteps = async () => {
      this.setState({loading: true});

      const authorized = await FitnessTracker.authorize(permissions);

      this.setState({loading: false});

      console.log(authorized, 'authorized!!!!');

      // if (!authorized) {
      //   return;
      // }

      try {
        const stepsWeekDaily = await FitnessTracker.getStatisticWeekDaily(
          FitnessDataType.Steps,
        );
        // const distanceWeekDaily = await FitnessTracker.getStatisticWeekDaily(
        //   FitnessDataType.Distance,
        // );

        // returns the number of steps walked today, e.g. 320
        console.log(stepsWeekDaily, 'stepsWeekDaily');
      } catch (e) {
        console.log(e);
      }
    };
```

**Comment by alarm109** [comment, marked as answer]
Hey @pandey21197,

Have you finished the [Android setup](https://kilohealth.github.io/rn-fitness-tracker/docs/fundamentals/getting-started#android)? Have you setup the OAuth screen? And do you get the response false after selecting email or after the OAuth screen?

**Reply by pandey21197** [reply]
@alarm109 yess done with Oauth screen and done with all the android setup. This "false" response iam getting post the OAuth screen.

**Reply by alarm109** [reply]
When in debugging, I think it doesn't take the firebase config settings, therefore you need to add debug.keystore sha1 to your google console credentials. To do so follow these instructions:
1. Go to [Google cloud console "Credentials" tab](https://console.cloud.google.com/apis/credentials)
2. At the top press button "**+ ADD CREDENTIALS**"
3. Select "**OAuth client ID**"
4. Application type choose "**Android**"
5. Give it a name
6. Add package name e.g. "**com.yourappname**". You can find your app package name in `AndroidManifest.xml` file.
7. Add SHA-1 signing certificate fingerprint. To get your SHA-1 run command: `keytool -keystore path-to-debug-or-production-keystore -list -v`, just change your `path-to-debug-or-production-keystore` to your path to keystore. E.g. `keytool -keystore android/app/debug.keystore -list -v`

After adding this credential it should work when running your app locally.

**Reply by pandey21197** [reply]
@alarm109 got it. I have actually did this step for my release keystore and iam running in debug mode. Understood, my bad 😅 Thanks for this lighting fast help 🙌
